core/src/main/scala/pureconfig/error/ConfigReaderFailure.scala

```diff
@@ -0,0 +1,18 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package pureconfig.error
+
+import com.typesafe.config.ConfigValue
+
+sealed abstract class ConfigReaderFailure
+final case object CannotConvertNull extends ConfigReaderFailure
```

Comment:
That distinction might be worth a comment to help differentiate the purpose of those two types.